modular_skyrat/modules/reagent_forging/code/crafting_bench.dm

```diff
@@ -0,0 +1,242 @@
+/obj/structure/reagent_crafting_bench
+	name = "forging work bench"
+	desc = "A crafting bench used to create forging items."
+	icon = 'modular_skyrat/modules/reagent_forging/icons/obj/forge_structures.dmi'
+	icon_state = "crafting_bench_empty"
+
+	anchored = TRUE
+	density = TRUE
+
+	///the current goal item that is obtainable
+	var/goal_item_path
+	///the name of the goal item
+	var/goal_name
+	///the amount of chains within the bench
+	var/chain_amount = 0
```

Comment:
This variable, and all the other _amount variables, would be better named as something like "current_[item]", so it firmly establishes "this is the CURRENT number of [item]"; as-is its hard to tell if "amount" is a goal, a current statistic, etc.